src/path/sentry_path_unix.c

```diff
@@ -25,6 +27,84 @@ struct sentry_pathiter_s {
     DIR *dir_handle;
 };
 
+static size_t
+write_loop(int fd, const char *buf, size_t buf_len)
+{
+    size_t offset = 0;
+    size_t remaining = buf_len;
+
+    while (true) {
+        if (remaining == 0) {
+            break;
+        }
+        ssize_t n = write(fd, buf + offset, remaining);
+        if (n <= 0) {
+            if (errno == EAGAIN || errno == EINTR) {
+                continue;
+            }
+            break;
+        }
+        offset += n;
+        remaining -= n;
+    }
+
+    return remaining;
+}
+
+bool
+sentry__filelock_try_lock(sentry_filelock_t *lock)
+{
+    int fd = open(lock->path->path, O_RDWR | O_CREAT | O_EXCL,
+        S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH);
+
+    if (fd < 0) {
+        if (errno == EEXIST) {
+            // check if the pid inside the lockfile still exists
+            char *contents = sentry__path_read_to_buffer(lock->path, NULL);
+            if (contents) {
+                int filepid = strtol(contents, NULL, 10);
+                bool process_exists
+                    = !(kill(filepid, 0) == -1 && errno == ESRCH);
```

Comment:
`strtol` returns `0` if the conversion fails. This would mean that the signal is sent to *every* process. Please check for `filepid == 0` explicitly and assume that someone has the lock. This can happen if the file was just created but not written to by another process.

This also shows a race condition. I think you need to require the lock *before* reading from it.